src/lib/src/charts/timeseries-chart.component.ts

```diff
@@ -97,37 +90,66 @@ export class TimeseriesChartComponent implements OnChanges {
       }
 
       if (this.formatX) {
-        config.axis.x['type'] = 'timeseries';
-        config.axis.x.tick['format'] = this.formatX;
+        config['axis'] = {
+          ...config['axis']
+        };
+        config['axis']['x'] = {
+          ...config['axis']['x'],
+          type: 'timeseries',
+          tick: {
+            format: this.formatX
+          }
+        };
+      }
+      // limit the max number of ticks, which honestly doesn't always work well with timeseries tick labels
+      if (this.maxTicks && this.datasets[0].data.length > this.maxTicks) {
+        config['axis'] = {
+          ...config['axis']
+        };
+        config['axis']['x'] = {
+          ...config['axis']['x'],
+          tick: {
+            ...config['axis']['x']['tick'],
+            count: this.maxTicks
+          }
+        };
+        if (this.type === 'bar') {
+          config['bar'] = {
```

Comment:
It's generally best to leave C3 config defaults alone unless you have good reason, so I've limited setting the bar chart width to only when we're > maxTicks. I'm using 48 for maxTicks and only with the hourly charts, which relates back to [metrics issue #229](https://github.com/PRX/metrics.prx.org/issues/229)